### DISCUSSION TITLE
Checking cache on model group fallbacks

### DISCUSSION THREAD

**Discussion by Manouchehri** [BerriAI/litellm#4212] [General]
### What happened?

When a call in an LLM group fails, it seems like the cache is checked again. This is pretty wasteful since it's highly unlikely that this will ever result in a cache hit.

<img width="773" alt="image" src="https://github.com/BerriAI/litellm/assets/7232674/a1e898a3-e726-4308-b642-cef4ff002993">

I would assume this is a bug across all caching providers, not just S3.

### Twitter / LinkedIn details

https://www.linkedin.com/in/davidmanouchehri/

**Comment by ghost** [comment]
>  it seems like the cache is checked again

it seems reasonable to check cache when the cache key changes 

we have optimizations around reducing the number of cache get requests - e.g. batching gets for a key - https://docs.litellm.ai/docs/proxy/caching#turn-on-batch_redis_requests

--

i don't view this as a bug, but i'm happy to convert to discussion and see what works best!
